### PR TITLE
Add borderless styleType to DropdownButton

### PR DESCRIPTION
**Pull request by mayank99**
Previously, we were omitting the `styleType` prop, but after discussion with @FlyersPh9 and @raplemie, it is a valid case to have a borderless dropdown, so I added it.

Also fixed an issue with partial props syntax in unit tests.

### CODE REVIEW

File: src/core/Buttons/DropdownButton/DropdownButton.test.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { MenuItem } from '../../Menu';
 import SvgCaretDownSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretDownSmall';
 import SvgCaretUpSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretUpSmall';
 
-function renderComponent(...props: Partial<DropdownButtonProps>[]) {
+function renderComponent(props?: Partial<DropdownButtonProps>) {
```
Comment (mayank99) [inline src/core/Buttons/DropdownButton/DropdownButton.test.tsx:13]: The previous syntax was not working, causing the test to fail with this error
image

Comment (mayank99) [reply src/core/Buttons/DropdownButton/DropdownButton.test.tsx:13]: I went through all instances of "Partial" and found the same in SplitButton so I fixed there too

File: src/core/Buttons/DropdownButton/DropdownButton.tsx [hunk 2]
```diff
@@ -19,7 +19,13 @@ export type DropdownButtonProps = {
...
    */
   menuItems: (close: () => void) => JSX.Element[];
-} & Omit<ButtonProps, 'onClick' | 'styleType' | 'endIcon'>;
+  /**
+   * Style of the dropdown button.
+   * Use 'borderless' to hide outline.
+   * @default 'default'
+   */
+  styleType?: 'default' | 'borderless';
+} & Omit<ButtonProps, 'onClick' | 'endIcon'>;
```
Comment (bentleyvk) [inline src/core/Buttons/DropdownButton/DropdownButton.tsx:28]: Omit `styleType` too:
image

File: src/core/Buttons/DropdownButton/DropdownButton.tsx [hunk 3]
```diff
@@ -19,7 +19,13 @@ export type DropdownButtonProps = {
    * and returns a list of `MenuItem` components.
    */
   menuItems: (close: () => void) => JSX.Element[];
-} & Omit<ButtonProps, 'onClick' | 'styleType' | 'endIcon'>;
+  /**
+   * Style of the dropdown button.
+   * Use 'borderless' to hide outline.
+   * @default 'default'
+   */
+  styleType?: 'default' | 'borderless';
```
Comment (bentleyvk) [inline src/core/Buttons/DropdownButton/DropdownButton.tsx:27]: When switching between style types menu with stays the same:
image
It's a minor thing. Strange but when changing other props in Storybook, menu width changes accordingly.

Comment (bentleyvk) [reply src/core/Buttons/DropdownButton/DropdownButton.tsx:27]: Oh, I found why 😀
image
We need to add `styleType` to deps list.

File: src/core/Buttons/DropdownButton/DropdownButton.test.tsx (hunk 1 above)
Comment (veekeys) [reply src/core/Buttons/DropdownButton/DropdownButton.test.tsx:13]: What, so it was not working all the time? How tests were passing then..

Comment (veekeys) [reply src/core/Buttons/DropdownButton/DropdownButton.test.tsx:13]: Oh ok.. We were not passing anything, so it was not complaining... Interesting

File: src/core/Buttons/DropdownButton/DropdownButton.tsx (hunk 3 above)
Comment (mayank99) [reply src/core/Buttons/DropdownButton/DropdownButton.tsx:27]: Yeah I noticed that, but is it really needed? Who is gonna change styleType in practice?

File: src/core/Buttons/DropdownButton/DropdownButton.tsx (hunk 2 above)
Comment (mayank99) [reply src/core/Buttons/DropdownButton/DropdownButton.tsx]: Done

File: src/core/Buttons/DropdownButton/DropdownButton.tsx (hunk 3 above)
Comment (bentleyvk) [reply src/core/Buttons/DropdownButton/DropdownButton.tsx:27]: But it is an easy fix and since we know that it will bug me 😀

Comment (mayank99) [reply src/core/Buttons/DropdownButton/DropdownButton.tsx:27]: Yeah that's why I already did it 😄

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**